libs/domain/site/mode-selector/src/mode-selector.component.ts

```diff
@@ -0,0 +1,114 @@
+import { AppRef, ComponentsPlugin, StorageService } from '@spryker-oryx/core';
+import { resolve } from '@spryker-oryx/di';
+import {
+  hydratable,
+  i18n,
+  isDefined,
+  subscribe,
+} from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+import { state } from 'lit/decorators.js';
+import { filter, tap } from 'rxjs';
+import { styles } from './mode-selector.styles';
+
+export const modeStorageKey = 'oryx.modeStorageKey';
+
+@hydratable('window:load')
+export class SiteModeSelectorComponent extends LitElement {
+  static styles = styles;
+
+  protected darkMode = 'mode-dark';
+  protected lightMode = 'mode-light';
+  protected root =
+    resolve(AppRef).findPlugin(ComponentsPlugin)?.getRoot() ?? 'body';
+  protected storage = resolve(StorageService);
+
+  @subscribe()
+  protected mode$ = this.storage.get<string>(modeStorageKey).pipe(
```

Comment:
just to be sure, this one currently will emit only once, it's not observing changes to the storage. We may improve it in the future, but now this subscription is technically dead, if first emission is undefined.